src/AutoMerge/guidelines.jl

```diff
@@ -857,6 +866,65 @@ function meets_version_can_be_imported(
     end
 end
 
+const guideline_linecounts_meet_thresholds = Guideline(;
+    info="Test that lines of source, tests and documentation meet specified thresholds",
+    docs="""Make sure that various line counts meet or exceed specified thresholds.
+Thresholds are controlled by these fields of GitHubAutoMergeData:
+ * src_min_lines:       Minimum number of lines of source code
+ * readme_min_lines:  % Minimum number of lines in the README file
+ * test_min_lines:    % Minimum number of lines of code in the test directory
+ * doc_min_lines:     # Minimum number of lines of documentation
+
+Those marked with a % can be expressed as a count, or as a fraction
+of the number of lines of source code.  For test_min_lines and
+doc_min_lines, the denominator of the fraction also includes the
+number of lines of test code.
+""",
+    check=(data, guideline_parameters) ->
+        linecounts_meet_thresholds(data.pkg_code_path, guideline_parameters)
+)
+
+function linecounts_meet_thresholds(pkg_code_path,
+                                    guideline_parameters)
+    analysis = PackageAnalyzer.analyze(pkg_code_path;)
+    # @info analysis
+    if isempty(analysis.lines_of_code)
+        return false, "PackageAnalyzer didn't find any lines of code."
+    end
+    # get parameters:
+    src_min_lines    = get(guideline_parameters, :src_min_lines,   10)
+    readme_min_lines = get(guideline_parameters, :readme_min_lines, 5)
+    test_min_lines   = get(guideline_parameters, :test_min_lines,   0.1f0)
+    doc_min_lines    = get(guideline_parameters, :doc_min_lines,    0.1f0)
+    issues = []
+    src_line_count = PackageAnalyzer.count_julia_loc(analysis, "src")
+    if !meets_threshold(readme_min_lines,
+                         PackageAnalyzer.count_readme(analysis),
+                         src_line_count)
+        push!(issues, "Package README file is too short")
```

Comment:
I think we should make these errors include the counts, so they are a bit more informative:
```suggestion
        push!(issues, "Package README file is too short. Counted $(src_line_count) lines, but at least $(readme_min_lines) lines are required.")
```